src/server/scripts/Kalimdor/zone_orgrimmar.cpp

```diff
@@ -263,8 +264,557 @@ class npc_thrall_warchief : public CreatureScript
+    SAY_SYLVANAS_1              = 1, // As the combined Horde and Alliance forces began their assault upon the Wrath Gate, an uprising broke out in the Undercity. Varimathras and hordes of his demonic brethren attacked. Hundreds of my people were slain in the coup. I barely managed to escape with my life.
+    
+    SAY_JAINA_0                 = 0, // Thrall, what has happened? The King is preparing for war...
+    SAY_JAINA_1                 = 1, // I will deliver this information to King Wrynn, Thrall, but...
+    SAY_JAINA_2                 = 2, // Bolvar was like a brother to him. In the King's absence, Bolvar kept the Alliance united. He found strength for our people in our darkest hours. He watched over Anduin, raising him as his own.
+    SAY_JAINA_3                 = 3, // I fear that the rage will consume him, Thrall. I remain hopeful that reason will prevail, but we must prepare for the worst... for war.
+    SAY_JAINA_4                 = 4 // Farewell, Warchief. I pray that the next time we meet it will be as allies.
+};
+
+enum HeraldEvents
+{
+    EVENT_HERALD_SCENE1         = 1,
+    EVENT_HERALD_SCENE2         = 2,
+    EVENT_HERALD_SCENE3         = 3,
+    EVENT_HERALD_SCENE4         = 4,
+    EVENT_HERALD_SCENE5         = 5,
+    EVENT_HERALD_SCENE6         = 6,
+    EVENT_HERALD_SCENE7         = 7,
+    EVENT_HERALD_SCENE8         = 8,
+    EVENT_HERALD_SCENE9         = 9,
+    EVENT_HERALD_SCENE10        = 10,
+    EVENT_HERALD_SCENE11        = 11,
+    EVENT_HERALD_SCENE12        = 12,
+    EVENT_HERALD_SCENE13        = 13,
+    EVENT_HERALD_SCENE14        = 14,
+    EVENT_HERALD_SCENE15        = 15,
+    EVENT_HERALD_RESET          = 16
+};
+
+Position const GuardsSpawnPosition[4] =
+{
+    { 1909.39f, -4144.21f, 40.6368f, 0.042239f },
+    { 1910.73f, -4155.26f, 40.6316f, 0.615577f },
+    { 1934.01f, -4141.40f, 40.6375f, 3.61109f },
+    { 1931.11f, -4156.38f, 40.6130f, 2.19737f }
+};
+
+Position const GuardsMovePosition[4] =
+{
+    { 1917.461670f, -4147.514160f, 40.636799f, 5.89346f },
+    { 1916.181274f, -4152.295898f, 40.629120f, 0.497757f },
+    { 1926.435425f, -4146.397461f, 40.618534f, 3.846709f },
+    { 1926.519165f, -4153.216797f, 40.614975f, 2.570434f }
+};
+
+Position const PortalsPosition[3] =
+{
+    { 1912.070923f, -4151.778809f, 40.637005f, 1.635784f }, // left
+    { 1931.545288f, -4149.240723f, 40.600616f, 1.804645f }, // right
+    { 1921.752441f, -4151.148438f, 40.623848f, 1.714324f }  // center
+};
+
+Position const MiscMovePositions[3] =
+{
+    { 1921.719604f, -4143.051270f, 40.623356f, 1.657789f }, // jaina move
+    { 1921.151855f, -4139.343750f, 40.583084f, 4.732627f }, // thrall move
+    { 1918.732422f, -4139.619629f, 40.607685f, 4.803311f } // sylvanas move
+};
+
+const uint32 GUARD_GUIDS_SIZE       = 4;
```

Comment:
this should be used as size for every array related to guards, like GuardsSpawnPosition and GuardsMovePosition